src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/RerankingTask.java

```diff
@@ -285,10 +295,34 @@ public Uni<RerankingTaskResult> get() {
                           "limit", limit,
                           "passages", passages))));
 
-      return rerankingProvider
-          .rerank(query.query(), passages, credentials)
+      // --- Record Passage Counts ---
+      final int passageCount = passages.size();
+      rerankingMetrics.recordTenantPassageCount(passageCount);
+      rerankingMetrics.recordAllPassageCount(passageCount);
+
+      // This executes the internal sync prep - no actual network call yet
+      Uni<RerankingProvider.RerankingResponse> rerankResponseUni =
```

Comment:
we can move this to be on line 311 because we know the sync prep work is no important, I was mostly concerned with when the actual call is made.